File: STRLineLoadConcentrated.m
classdef STRLineLoadConcentrated < handle

    properties
        Id;
        LoadCaseId;
        Fx;
        Fy;
        Fz;
        Mx;
        My;
        Mz;
        AppliedTo;
        RelativeLocation;
    end

    methods
        function obj = STRLineLoadConcentrated(id,LoadCaseId, fx, fy, fz, mx ,my , mz, relativeLocation)
                obj.Id = id;
                obj.LoadCaseId = LoadCaseId;
                obj.Fx = fx;
                obj.Fy = fy;
                obj.Fz = fz;
                obj.Mx = mx;
                obj.My = my;
                obj.Mz = mz;
                obj.RelativeLocation = relativeLocation;
        end

        function ToString(obj)
            fprintf("Line Load Concentrated #%i LC%i @(%5.2f)\n", obj.Id,obj.LoadCaseId, obj.RelativeLocation);
            fprintf("Fx = %5.2f\t", obj.Fx);
            fprintf("Fy = %5.2f\t", obj.Fy);
            fprintf("Fz = %5.2f\n", obj.Fz);
            fprintf("Mx = %5.2f\t", obj.Mx);
            fprintf("My = %5.2f\t", obj.My);
            fprintf("Mz = %5.2f\n", obj.Mz);
            fprintf("Applied To ==> %i\n", obj.AppliedTo);
        end
    end
end

File: src/STRLineLoadConcentrated.cpp
#include "STRLineLoadConcentrated.hpp"
#include <iostream>
#include <vector>

STRLineLoadConcentrated::STRLineLoadConcentrated(int Id, int LoadCaseId, double Fx, double Fy, double Fz,
	double Mx, double My, double Mz, double RL)
	: Id(Id), LoadCaseId(LoadCaseId), Fx(Fx), Fy(Fy), Fz(Fz), Mx(Mx), My(My), Mz(Mz) ,RelativeLoacation(RL){
}


void STRLineLoadConcentrated::ToString() const {
	std::cout << "Line Load Concentrated ID: " << Id << ", Load Case ID: " << LoadCaseId << "@Relative Location: " << RelativeLoacation  << std::endl;
	std::cout << "Forces: Fx=" << Fx << ", Fy=" << Fy << ", Fz=" << Fz << std::endl;
	std::cout << "Moments: Mx=" << Mx << ", My=" << My << ", Mz=" << Mz << std::endl;
	std::cout << "Applied To Elements: ";
	for (size_t i = 0; i < AppliedTo.size(); ++i) {
		std::cout << AppliedTo[i];
		if (i < AppliedTo.size() - 1) {
			std::cout << ", ";
		}
	}
	std::cout << std::endl;
}
